sim-lib/src/lib.rs

```diff
@@ -886,14 +891,19 @@ async fn write_payment_results(
     mut receiver: Receiver<(Payment, PaymentResult)>,
     listener: Listener,
     print_batch_size: u32,
+    no_results: bool,
 ) -> Result<(), SimulationError> {
-    let mut writer = WriterBuilder::new().from_path(format!(
-        "simulation_{:?}.csv",
-        SystemTime::now()
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs()
-    ))?;
+    let mut writer = if no_results {
+        WriterBuilder::new().from_path("/dev/null")?
```

Comment:
Why not have an `Option` here and skip it all-together below so that we don't have to write anything to disk?

Reply:
I played around with that approach yesterday and it is viable, it just adds more boilerplate.

I'll add a commit on top of this with that approach so you guys can compare it. We can either squash it or remove it depending on what you think

Reply:
Looks like the rust wizards have been on my side today, I was able to make this pretty minimal. Check 
365746b

Reply:
Let's squash it - looks good 😎